### PR TITLE
Address -Wunused-but-set-variable

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Several packages have variables that probably served for debugging purposes. 
Not all packages are treated but there are some in assertion code being modified in another PR.

<details><summary><h2>Note about Eigen</h2> (click for details)</summary>

There are also warnings concerning Eigen and we might make a merge request.
**Edit:** the merge request was [created and is already merged](https://gitlab.com/libeigen/eigen/-/merge_requests/1094).

[SparseCore/TriangularSolver.h](https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/SparseCore/TriangularSolver.h#L279)

`/usr/include/eigen3/Eigen/src/SparseCore/TriangularSolver.h:273:13: warning: variable 'count' set but not used [-Wunused-but-set-variable]`

[SparseLU/SparseLU_heap_relax_snode.h](https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/SparseLU/SparseLU_heap_relax_snode.h#L80)
`/usr/include/eigen3/Eigen/src/SparseLU/SparseLU_heap_relax_snode.h:79:9: warning: variable 'nsuper_et' set but not used [-Wunused-but-set-variable]`

and the same for the other variable `nsuper_et_post`.  The increments must be removed as well.

See for example https://cgal.geometryfactory.com/CGAL/testsuite/results-5.6-I-86.shtml#Heat_method_3 at the [column `Debian-testing-clang-main`](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.6-I-86/Heat_method_3/TestReport_gimeno_Debian-testing-clang-main.gz).

</details>

## Release Management

* Affected package(s): several

* License and copyright ownership: unchanged

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h [hunk 1]
```diff
@@ -906,8 +915,9 @@ namespace internal {
         put(degree, vc, vvc);
         put(degree, vd, vvd);
 
+#ifdef CGAL_PMP_REMESHING_VERBOSE
         ++nb_flips;
-
+#endif
 #ifdef CGAL_PMP_REMESHING_VERBOSE_PROGRESS
         std::cout << "\r\t(" << nb_flips << " flips)";
```
Comment (MaelRL) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h:922]: why not use the existing macro? `nb_flips` and `nb_collapses` are defined behind `CGAL_PMP_REMESHING_VERBOSE` and used behind `CGAL_PMP_REMESHING_VERBOSE_PROGRESS`.

Comment (afabri) [reply Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h:922]: I was wondering it made sense to also write the lower case versions, that is functions.

Comment (MaelRL) [reply Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h:922]: > The more complicated macros call `#define CGAL_PMP_REMESHING_VERBOSE`

That's not what I meant; I thought it was only using within the `_PROGRESS` block but I hadn't seen there is a call at the end:
```
#ifdef CGAL_PMP_REMESHING_VERBOSE
      std::cout << "\r\tdone ("<< nb_flips << " flips)" << std::endl;
#endif
```
Sorry!

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-99

File: Mesh_3/include/CGAL/Mesh_3/search_for_connected_components_in_labeled_image.h [hunk 2]
```diff
@@ -137,7 +137,9 @@ search_for_connected_components_in_labeled_image(const CGAL::Image_3& image,
             {
               visited[offset] = true;
               second_pass[offset] = false;
+#if CGAL_MESH_3_SEARCH_FOR_CONNECTED_COMPONENTS_IN_LABELED_IMAGE_VERBOSE > 1
               ++nb_voxels;
+#endif
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/search_for_connected_components_in_labeled_image.h:142]: This is not correct. The `nb_voxels` variable is used here:
https://github.com/afabri/cgal/blob/8674744f1719ced37df9ef566ea709ec603b9a42/Mesh_3/include/CGAL/Mesh_3/search_for_connected_components_in_labeled_image.h#L228-L239

```suggestion
#ifdef CGAL_MESH_3_SEARCH_FOR_CONNECTED_COMPONENTS_IN_LABELED_IMAGE_VERBOSE
              ++nb_voxels;
#endif
```

**Pull request merged**